.github/actions/check_cirrus_cron/cron_failures.sh

```diff
@@ -0,0 +1,119 @@
+
+err() {
+    # Ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions
+    echo "::error file=${BASH_SOURCE[0]},line=${BASH_LINENO[0]}::${1:-No error message given}"
+    exit 1
+}
+
+if [[ -z "$GITHUB_REPOSITORY" ]]; then
+    err "Expecting \$GITHUB_REPOSITORY value to not be empty"
+elif [[ ! -w "$GITHUB_ENV" ]]; then
+    err "Expecting \$GITHUB_ENV ($GITHUB_ENV) to be path to writable file"
+fi
+
+mkdir -p artifacts
+cat > ./artifacts/query_raw.json << "EOF"
+{"query":"
+  query CronNameStatus($owner: String!, $repo: String!) {
+    githubRepository(owner: $owner, name: $repo) {
+      cronSettings {
+        name
+        lastInvocationBuild {
+          id
+          status
+        }
+      }
+    }
+  }
+",
+"variables":"{
+  \"owner\": \"@@OWNER@@\",
+  \"repo\": \"@@REPO@@\"
+}"}
+EOF
+# Makes for easier copy/pasting query to/from
+# https://cirrus-ci.com/explorer
+owner=$(cut -d '/' -f 1 <<<"$GITHUB_REPOSITORY")
+repo=$(cut -d '/' -f 2 <<<"$GITHUB_REPOSITORY")
+sed -i -r -e "s/@@OWNER@@/$owner/g" -e "s/@@REPO@@/$repo/g" ./artifacts/query_raw.json
+
+echo "::group::Posting GraphQL Query"
+# Easier to debug in error-reply when query is compacted
+tr -d '\n' < ./artifacts/query_raw.json | tr -s ' ' | tee ./artifacts/query.json | \
+    jq --indent 4 --color-output .
+
+if grep -q '@@' ./artifacts/query.json; then
+    err "Found unreplaced substitution token in raw query JSON"
+fi
+curl \
+  --request POST \
+  --silent \
+  --location \
+  --header 'content-type: application/json' \
+  --url 'https://api.cirrus-ci.com/graphql' \
+  --data @./artifacts/query.json \
+  --output ./artifacts/reply.json
+echo "::endgroup::"
+
+echo "::group::Received GraphQL Reply"
+jq --indent 4 --color-output . <./artifacts/reply.json || \
+    cat ./artifacts/reply.json
+echo "::endgroup::"
+
+# Desireable to catch non-JSON encoded errors in reply.
+if grep -qi 'error' ./artifacts/reply.json; then
+    err "Found the word 'error' in reply"
+fi
+
+# e.x. reply.json
+# {
+#   "data": {
+#     "githubRepository": {
+#       "cronSettings": [
+#         {
+#           "name": "Keepalive_v2.0",
+#           "lastInvocationBuild": {
+#             "id": "5776050544181248",
+#             "status": "EXECUTING"
+#           }
+#         },
+#         {
+#           "name": "Keepalive_v1.9",
+#           "lastInvocationBuild": {
+#             "id": "5962921081569280",
+#             "status": "COMPLETED"
+#           }
+#         },
+#         {
+#           "name": "Keepalive_v2.0.5-rhel",
+#           "lastInvocationBuild": {
+#             "id": "5003065549914112",
+#             "status": "FAILED"
+#           }
+#         }
+#       ]
+#     }
+#   }
+# }
+_filt='[.data.githubRepository.cronSettings | map(select(.lastInvocationBuild.status=="FAILED")) | .[].name, .[].lastInvocationBuild.id] | join(" ")'
+jq --raw-output "$_filt" ./artifacts/reply.json > ./artifacts/name_id.csv
```

Comment:
Could I suggest a different name? The `.csv` extension greatly confused me when I got to the `read` below. Likewise with the other csv file, `cron_fail_mail.csv`.

Reply:
Could you clarify please, I'm confused over your focus on `.csv`.  That extension is (sometimes) used for space-separated files, the `,` delimiter isn't the important part.  Do you have a suggestion for a better name?

Note: In the case of `cron_fail_mail.csv`...this is actually a traditional CSV file.  The github-action accepts multiple "to" addresses, which I would like to utilize (if it weren't for the spamhaus SBL SNAFU).

Reply:
I'veen `.tsv` for tab-separated, and I've seen an option in Libreoffice for selecting the field delimiter when importing `.csv` files, but in all my years I've never seen anyone actually use anything except comma in a CSV. It feels wrong. I went to [Wikipedia](https://en.wikipedia.org/wiki/Comma-separated_values) to see if I'm just too old, but their paragraph on alternative delimiters is chock full of `[citation needed]` and `who does this?`. I'd be OK with `.txt`, even `.dat`. I'd be OK with `.tsv` and using tabs. Or `.csv` with commas. But `.csv` with spaces is just too counterintuitive; it will cause headaches for anyone who has to read this in the future.

>Note: In the case of cron_fail_mail.csv...this is actually a traditional CSV file. The github-action accepts multiple "to" addresses, which I would like to utilize (if it weren't for the spamhaus SBL SNAFU).

Having multiple values doesn't make it CSV: in this case I dare say that the file contains one single field, very probably read in one gulp, and that field is a comma-separated list of email addresses. Subtle distinction, but important IMHO: CSV suggests that there will be multiple rows, each with a constant number of columns. If I understand this file correctly, it's one single row, containing one single field.

Reply:
Renamed to `cron-fail_addrs.csv` to be more clear.

I tried expanding `name_id.csv` more verbosely, but all my attempts result in an unwieldy/long fugly names.  I think this is something we can leave to be fixed later if necessary.

Reply:
Sorry, I don't understand this comment; what does "expanding name_id.csv more verbosely" mean? I was just asking for a rename to "name_id.txt"

Reply:
oh okay, ya, that's what I was asking about initially.  I got it now.